src/Adapter/Product/Update/ProductStockUpdater.php

```diff
@@ -156,15 +153,13 @@ private function updateStockAvailable(Product $product, StockAvailable $stockAva
     /**
      * @param Product $product
      * @param StockAvailable $stockAvailable
-     * @param array $propertiesToUpdate
-     * @param bool $addMovement
      */
-    private function updateQuantity(Product $product, StockAvailable $stockAvailable, array $propertiesToUpdate, bool $addMovement): void
+    private function updateQuantity(Product $product, StockAvailable $stockAvailable): void
     {
         $deltaQuantity = (int) $product->quantity - (int) $stockAvailable->quantity;
         $stockAvailable->quantity = (int) $product->quantity;
 
-        if ($addMovement && 0 !== $deltaQuantity) {
+        if (0 !== $deltaQuantity) {
```

Comment:
```suggestion
        if (0 < $deltaQuantity) {
```

Reply:
No the delta can be negative when you reduce the quantity